docs/api/csf/csf-factories.mdx

```diff
@@ -0,0 +1,328 @@
+
+### Codemod
+
+Storybook provides codemods to help you migrate your codebase to the CSF Factories format effortlessly. Run the following command in the root of your project: 
+
+```sh
+npx storybook automigrate csf-factories
+```
+
+<Callout variant="info" icon="💡">
+  If you use a monorepo or your `.storybook` directory is not at the root, make sure to pass the `--config-dir` flag to the command, specifying its path.
+</Callout>
+
+This command performs the following actions:
+
+- Adds [a subpath import](#1-add-subpath-import-in-packagejson) for the preview file in your `package.json`
+- [Migrates `.storybook/main.js`](#2-update-your-main-storybook-config-file)
+- [Migrates `.storybook/preview.js`](#3-update-your-preview-config-file)
+- Converts all story files to [use CSF Factories](#4-update-your-story-files)
+
+This should get you started with CSF Factories immediately. If you prefer a manual migration, the next sections detail the necessary updates.
+
+### 1. Add subpath import in `package.json`
+
+To be able to consistently import the preview file from any location in your project, you need to add a subpath import in your `package.json`. For more information, refer to the [subpath imports documentation](../../writing-stories/mocking-data-and-modules/mocking-modules.mdx#subpath-imports).
+
+{/* TODO: Snippetize */}
+```json title="package.json"
+{
+  "imports": {
+    "#.storybook/preview": {
+      "default": "./.storybook/preview.ts",
+    },
+    "#*": ["./*", "./*.ts", "./*.tsx"],
+  },
+}
+```
+
+### 2. Update your main Storybook config file
+
+Update your `.storybook/main.ts` file to use the new [`defineMain`](#definemain) function.
+
+```diff title=".storybook/main.ts"
+// Replace your-framework with the framework you are using (e.g., react-vite, nextjs, experimental-nextjs-vite)
++ import { defineMain } from '@storybook/your-framework/node';
+- import { StorybookConfig } from '@storybook/your-framework';
+
++ export default defineMain({
+- export const config: StorybookConfig = {
+    // ...current config
++ });
+- };
+- export default config;
+```
+
+### 3. Update your preview config file
+
+Update your `.storybook/preview.ts` file to use the new [`definePreview`](#definepreview) function.
+
+<details>
+<summary>Which addons should be specified in `preview`?</summary>
+
+The ability for an addon to provide annotation types (`parameters`, `globals`, etc.) is new and not all addons support it yet. 
+
+If an addon provides annotations (i.e. it distributes a `./preview` export), it can be imported in two ways:
+
+1. For official Storybook addons, you import the default export:
+   `import addonName from '@storybook/addon-name'`
+
+2. For community addons, you should import the entire module and access the addon from there:
+   `import * as addonName from 'community-addon-name'`
+
+Sound complicated? The [codemod](#codemod) will automatically add the necessary imports for you.   
+
+</details>
+
+```diff title=".storybook/preview.ts"
```

Comment:
minor item here for consistency, add js as well, turning this into `storybook/preview.js|ts`) and apply it to other places as well